ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/TestData.kt

```diff
@@ -104,6 +114,42 @@ internal object TestData {
     }
 }
 
+internal class MockApplicationContext: ApplicationContext {
+    override fun getApplicationName(): String {
+        return "Mock Paywall"
+    }
+}
+
+internal class MockViewModel(
+    private val offering: Offering
+): ViewModel(), PaywallViewModel {
+    override val state: StateFlow<PaywallViewState>
+        get() = _state.asStateFlow()
+    private val _state = MutableStateFlow(offering.toPaywallViewState(VariableDataProvider(MockApplicationContext())))
+
+    override fun refreshState() {}
+
+    override fun selectPackage(packageToSelect: TemplateConfiguration.PackageInfo) {
+        _state.value = when (val currentState = _state.value) {
+            is PaywallViewState.Loaded -> {
+                currentState.copy(selectedPackage = packageToSelect)
+            }
+            else -> {
+                Logger.e("Unexpected state trying to select package: $currentState")
+                currentState
+            }
+        }
```

Comment:
Hmm should we just fail here? I guess this might allow you to click on interactive mode... but might be easier to just change the test data if we need to change the package

Reply:
Good idea 👍🏻

Reply:
Done.